recipes/tiledb/all/conanfile.py

```diff
@@ -0,0 +1,267 @@
+            "Visual Studio": "16",
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        if self.options.serialization:
+            self.options["libcurl"].with_zstd = True
+        if self.options.s3:
+            self.options["aws-sdk-cpp"].s3 = True
+            self.options["aws-sdk-cpp"]["identity-management"] = True
+            self.options["aws-sdk-cpp"].sts = True
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        # TileDB has no transitive header deps
+        self.requires("bzip2/1.0.8")
+        self.requires("libxml2/2.12.5")
+        self.requires("lz4/1.9.4")
+        self.requires("spdlog/1.13.0")
+        self.requires("xz_utils/5.4.4")
+        self.requires("zlib/[>=1.2.11 <2]")
+        self.requires("zstd/1.5.5")
+        if self.settings.os != "Windows":
+            self.requires("openssl/[>=1.1 <4]")
+        self.requires("libmagic/5.45")
+        if self.options.azure:
+            self.requires("azure-storage-cpp/12.6.1")
+        if self.options.gcs:
+            self.requires("google-cloud-cpp/2.19.0")
+        if self.options.serialization:
+            self.requires("capnproto/1.0.1")
+            self.requires("libcurl/[>=7.78.0 <9]")
+        if self.options.s3:
+            self.requires("aws-sdk-cpp/1.11.160")
+        if self.options.tools:
+            self.requires("clipp/1.2.3")
+        if self.options.webp:
+            self.requires("libwebp/1.3.2")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+        if self.options.serialization and not self.dependencies["libcurl"].options.with_zstd:
+            raise ConanInvalidConfiguration("TileDB serialization requires libcurl with with_zstd option enabled.")
+        if self.options.s3:
+            aws_opts = self.dependencies["aws-sdk-cpp"].options
+            if not (aws_opts.get_safe("s3") and aws_opts.get_safe("identity-management") and aws_opts.get_safe("sts")):
+                raise ConanInvalidConfiguration(
+                    f"TileDB S3 support requires aws-sdk-cpp with 's3', 'identity-management' and 'sts' options enabled."
+                )
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.21 <4]")
+        if not self.conf.get("tools.gnu:pkg_config", default=False, check_type=str):
+            self.tool_requires("pkgconf/2.1.0")
```

Comment:
```suggestion
            self.tool_requires("pkgconf/2.2.0")
```

Reply:
You can now use a version range as well: `pkgconf/[>=2.2.0 <3]`.